hydra-android-gradle-plugin/src/main/java/com/pandora/hydra/android/HydraAndroidPlugin.java

```diff
@@ -45,28 +45,23 @@
 
     @Override
     public void apply(Project project) {
-        if (project.getSubprojects().isEmpty()) {
-            project.getLogger().info("Applying to leaf project: " + project.getName());
-            HydraPluginExtension hydraExtension = project.getExtensions().create("hydra", HydraPluginExtension.class);
-            project.afterEvaluate(p -> {
-                if(hydraExtension.isBalanceThreads()) {
-                    project.getLogger().info("Hydra Android doesn't support thread balancing. Ignoring setting");
-                    hydraExtension.setBalanceThreads(false);
-                }
+        project.getLogger().info("Applying to project: " + project.getName());
+        HydraPluginExtension hydraExtension = project.getExtensions().create("hydra", HydraPluginExtension.class);
+        project.afterEvaluate(p -> {
+            if(hydraExtension.isBalanceThreads()) {
+                project.getLogger().info("Hydra Android doesn't support thread balancing. Ignoring setting");
+                hydraExtension.setBalanceThreads(false);
+            }
 
-                BalancedTestFactory<AndroidBalancedTest, AndroidUnitTest> factory = new BalancedTestFactory<>(AndroidBalancedTest.class,
-                        AndroidUnitTest.class,
-                        (balancedTest, originalTest) -> {
-                            balancedTest.setMergedManifest(originalTest.getMergedManifest());
-                            balancedTest.setResCollection(originalTest.getResCollection());
-                            balancedTest.setSdkPlatformDirPath(originalTest.getSdkPlatformDirPath());
-                        });
-                factory.createBalancedTest(p, hydraExtension);
-            });
-        } else {
-            project.getLogger().lifecycle("###########Pandora Hydra plugin should be applied at Android application leaf project level, which should not have any sub projects. Please use Hydra core plugin for library projects or tree of projects");
-            throw new RuntimeException("Pandora Hydra plugin should be applied at Android application leaf projects level, which should not have any sub projects");
-        }
+            BalancedTestFactory<AndroidBalancedTest, AndroidUnitTest> factory = new BalancedTestFactory<>(AndroidBalancedTest.class,
+                    AndroidUnitTest.class,
+                    (balancedTest, originalTest) -> {
+                        balancedTest.setMergedManifest(originalTest.getMergedManifest());
+                        balancedTest.setResCollection(originalTest.getResCollection());
+                        balancedTest.setSdkPlatformDirPath(originalTest.getSdkPlatformDirPath());
+                    });
+            factory.createBalancedTest(p, hydraExtension, false);
```

Comment:
Should this be configurable via the plugin extension, or do we really always want it to be false for android projects?

Reply:
I think it has to be this way until/unless we can figure out how to recursively apply Hydra to all Android modules in a project. When I first started looking at this, I think we were using the same logic as we were for non-Android projects, which doesn't work because we (Hydra plugin) were always attempting to apply the Android _application_ plugin to _all_ modules, which is invalid since many modules will require the Android _library_ plugin instead. However, with the work you and Alex did to support both Android and Java from the same Hydra codebase, we should revisit this area to see if we can somehow support recursive application for Android. I think that's out of scope for this PR though.

Reply:
Ok, sounds reasonable